File: app/src/main/java/io/github/pshegger/infobook/fragments/ExampleFragment.java
package io.github.pshegger.infobook.fragments;

import android.graphics.Color;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * Created by pshegger on 2015.10.04..
 */
public class ExampleFragment extends NamedFragment {
    private static final int[] colors = new int[] {
            Color.RED, Color.YELLOW, Color.GREEN,
            Color.BLUE, Color.CYAN
    };
    private static int idCtr = 0;

    private int myId;
    private String baseName;

    public static ExampleFragment newInstance() {
        return newInstance("Example");
    }

    public static ExampleFragment newInstance(String name) {
        ExampleFragment ef = new ExampleFragment();

        ef.myId = idCtr++;
        ef.baseName = name;

        return ef;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = new View(container.getContext());
        v.setBackgroundColor(colors[myId % colors.length]);

        return v;
    }

    @Override
    public String getName() {
        return baseName+myId;
    }
}

File: app/src/main/kotlin/io/github/pshegger/infobook/fragments/ExampleFragment.kt
package io.github.pshegger.infobook.fragments

import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

/**
 * Created by pshegger on 2015. 11. 10..
 */
class ExampleFragment : BaseInfoBookFragment() {
    companion object {
        var idCtr: Int = 0

        fun newInstance(name: String): ExampleFragment {
            val fragment = ExampleFragment()

            fragment.myId = idCtr++
            fragment.baseName = name

            return fragment
        }
    }

    private val colors = arrayOf(
            Color.RED, Color.YELLOW, Color.GREEN,
            Color.BLUE, Color.CYAN
    )

    var baseName: String = ""
    var myId: Int = 0


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val v = View(container?.context)
        v.setBackgroundColor(colors[myId.mod(colors.size)])

        return v
    }

    override fun name(): String = "$baseName$myId"
}